### PR TITLE
Add Item Prices plugin for item price/value tooltips

### PR DESCRIPTION
**Pull request by ChaoticConundrum**
Options are added to the mouse highlight plugin to display exchange prices and/or high alch values on mouse tooltips over items in the inventory and bank. (options are off by default)

screenshot_20180311_041018

The value-based color highlights are ripped from the ground items plugin, including color config options. It may be useful in the future to consolidate these options somewhere so other plugins can use the same options. I know there was at least one more PR that did some value-based highlighting.

This also includes refactoring of the mouse highlight overlay render method to use a switch on the first menu action type to determine whether tooltips should be drawn (and if anything should be added to them). I think this should be easier to maintain / add to in the future.

Also, while I was there, I removed tooltips from the bank pin window, logout window, world switcher and non-transparent chat box. I don't believe these tooltips added anything to these interfaces, and were just clutter.

### CODE REVIEW

**Comment by deathbeam** [comment]
As i mentioned on Discord earlier, I think the highlighting of price here is not needed at all, it makes sense for ground items, but for tooltips where you don't need to visually differentiate between items based on color, not much.

Also, the price should probably be separate tooltip and not included in item name tooltip (see item stats plugin for example).

**Comment by ChaoticConundrum** [comment]
It's easy to remove the colors.
I like the separate tooltip idea. I tried plugging that in, and it seems the TooltipOverlay breaks with more than two tooltips (normal tooltip + item stats + price). I could fix that, but if I do it that way, the question is should it be part of the mouse highlight plugin, item stats, or a new plugin?

**Comment by deathbeam** [comment]
Yeah, as the colors are not very useful there, it is imo better to get rid of them, that will also solve the issue with having duplicate configuration for ground items and here.
It is interesting that TooltipOverlay breaks, yeah, fixing that would be definitely good idea.
This plugin definitely do not belongs to item stats, so maybe keeping it in mouse highlighting plugin is okay? Although it was pretty simple plugin and this makes it a bit complicated, maybe even separate plugin will be good.

**Comment by ChaoticConundrum** [comment]
Alright, so refactored this into a new plugin. Now creates a new tooltip instead of adding to the mouse highlight tooltip. Removed colors. Also added the option to show price each on stacks.

screenshot_20180314_142030

Fixed an issue in TooltipOverlay to render more than two tooltips correctly. Now looks like:

screenshot_20180314_142618

To get the mouse highlight tooltip to always be on the bottom, added TooltipManager.addFront(). The additional tooltips appear in "undefined" order (probably the order the plugins are loaded; alphabetically). More work would be needed to make the tooltips appear in a defined order.

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/tooltip/TooltipOverlay.java [hunk 1]
```diff
@@ -88,14 +89,26 @@ public Dimension render(Graphics2D graphics, Point parent)
 				position.setLocation(tooltip.getPosition());
 			}
 
-			if (lastLocation.contains(position))
+			// check if this tooltip would overlap the last
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/tooltip/TooltipOverlay.java]: This looks like it is fixing a bug with the tooltip renderer? It should be in a separate commit with a proper commit message

File: runelite-client/src/main/java/net/runelite/client/plugins/itemprices/ItemPricesOverlay.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+				return stackValueText(qty, 0, Math.round(item.getPrice() * HIGH_ALCHEMY_CONSTANT));
+			}
+		}
+		return null;
+	}
+
+	private String stackValueText(int qty, int gePrice, int haValue)
+	{
+		itemStringBuilder.append("<col=eeeeee>");
+		if (qty <= 1)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemprices/ItemPricesOverlay.java]:
```java
			if (qty > 1 && (gePrice > 0 || haValue > 0))
			{
				itemStringBuilder.append("(").append(qty).append(") ");
			}
			if (gePrice > 0)
			{
				itemStringBuilder.append("EX: ")
						.append(StackFormatter.quantityToStackSize(gePrice * qty))
						.append(" gp");
				if (qty > 1 && config.showEA())
				{
					itemStringBuilder.append(" (")
							.append(StackFormatter.quantityToStackSize(gePrice))
							.append(" ea)");
				}
			}
			if (haValue > 0)
			{
				itemStringBuilder.append(gePrice > 0 ? ", " : "")
						.append("HA: ")
						.append(StackFormatter.quantityToStackSize(haValue * qty))
						.append(" gp");
				if (qty > 1 && config.showEA())
				{
					itemStringBuilder.append(" (")
							.append(StackFormatter.quantityToStackSize(haValue))
							.append(" ea)");
				}
			}
```

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/tooltip/TooltipManager.java [hunk 3]
```diff
@@ -42,6 +41,11 @@ public void add(Tooltip tooltip)
 		tooltips.add(tooltip);
 	}
 
+	public void addFront(Tooltip tooltip)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/tooltip/TooltipManager.java]: Probably put this in a separate commit aswell

File: runelite-client/src/main/java/net/runelite/client/plugins/itemprices/ItemPricesOverlay.java (hunk 2 above)
Comment (ChaoticConundrum) [reply runelite-client/src/main/java/net/runelite/client/plugins/itemprices/ItemPricesOverlay.java]: Good catch, that's definitely unecessary

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/tooltip/TooltipOverlay.java (hunk 1 above)
Comment (ChaoticConundrum) [reply runelite-client/src/main/java/net/runelite/client/ui/overlay/tooltip/TooltipOverlay.java]: Done
